src/Components/LifecycleChart/lifecycleChart.tsx

```diff
@@ -0,0 +1,131 @@
+
+
+const LifecycleChart: React.FC<React.PropsWithChildren> = () => {
+
+    // Start = y0, end = y
+    const alerts = [
+      [
+        {x: "Node.js 16", y0: new Date("2023-01"), y: new Date("2024-06"), packageType: 'Retired' }
+      ],
+      [
+        {x: "gcc-toolset 12", y0: new Date("2023-01"), y: new Date("2025-10"), packageType: 'Support ends within 6 months' }
+      ],
+      [
+        {x: "Ruby 3.1", y0: new Date("2024-08"), y: new Date("2025-06"), packageType: 'Not installed' }
+      ],
+      [
+        {x: "gcc-toolset 12", y0: new Date("2023-01"), y: new Date("2027-10"), packageType: 'Supported' }
+      ],
+      [
+        {x: "Ruby 3.0", y0: new Date("2024-08"), y: new Date("2025-06"), packageType: 'Upcoming release' }
+      ],      
+    ];
+
+
+
+    const formatDate = (date: Date, isTime: boolean) => {
+      const dateString = date?.toLocaleDateString("en-US", { year: 'numeric' });
+      const timeString = date?.toLocaleTimeString("en-US", { hour12: false });
+      return isTime ? `${dateString} ${timeString}` : dateString;
+    };
+
+    const getChart = (alert: any[], index: number) => {
+      const data: any[] = [];
+
+      alert?.map((datum: { packageType: string; }) => {
+        data.push({
+          ...datum,
+          x: alerts.length - index,
+          fill: datum.packageType === "Retired"
+            ? "var(--pf-v5-global--danger-color--100)"
+            : datum.packageType === "Support ends within 6 months"
+              ? "var(--pf-v5-global--warning-color--100)"
+              : datum.packageType === "Not installed"
+                ? "var(--pf-v5-global--palette--blue-200)"
+                : datum.packageType === "Supported"
+                  ? "var(--pf-v5-global--success-color--100)"
+                    : datum.packageType === "Upcoming release"
+                      ? "var(--pf-v5-global--palette--blue-100)"
+                      : "var(--pf-v5-global--default-color--300)",
+        });
+      });
+
+      if (data?.length === 0) {
+        return null;
+      }
+      return (
+        <ChartBar
+          data={data}
+          key={index}
+          style={{
+            data: {
+              fill: ({ datum }) => datum.fill,
+              stroke: ({ datum }) => datum.fill,
+            }
+          }}
+        />
+      );
+    };
+
+    return (
+      <Card>
+        <div style={{ height: "400px", width: "450px" }}>
+          <Chart
+            ariaDesc="Support timelines of packages and RHEL versions"
+            ariaTitle="Lifecycle bar chart"
+            containerComponent={
+              <ChartVoronoiContainer
+                labelComponent={
+                  <ChartTooltip constrainToVisibleArea labelComponent={<ChartLabel dx={-65} textAnchor="start" />} />
+                }
+                labels={({ datum }) => `Package: ${datum.x}\npackageType: ${datum.packageType}\nStart: ${formatDate(new Date(datum.y0), true)}\nEnd: ${formatDate(new Date(datum.y), true)}`}
```

Comment:
It looks like these are getting cut off/not being formatted in a way that makes a lot of sense. I don't think it's worth holding this PR over, but I will create a Jira ticket to address this.

Reply:
Yes the chart has a few bugs to work out